src/sentry/services/hybrid_cloud/organization/service.py

```diff
@@ -16,10 +18,30 @@
     ByOrganizationId,
     ByOrganizationIdAttribute,
     ByOrganizationSlug,
+    RegionResolution,
     UnimplementedRegionResolution,
 )
 from sentry.services.hybrid_cloud.rpc import RpcService, regional_rpc_method
 from sentry.silo import SiloMode
+from sentry.types.region import Region
+
+
+class ByOrganizationMembership(RegionResolution):
+    def resolve(self, arguments: ArgumentDict) -> Collection[Region]:
+        from sentry.models import OrganizationMemberMapping
+
+        user_id: int = arguments["user_id"]
+        member_mappings = OrganizationMemberMapping.objects.filter(user_id=user_id)
+        organization_ids = member_mappings.values_list("organization_id", flat=True)
+        mappings = self.organization_mapping_manager.filter(organization_id__in=organization_ids)
+        return [self._resolve_from_mapping(mapping) for mapping in mappings]
```

Comment:
Just curious, do we do de-duplication on the list of regions?

Reply:
Oh right, good point, it should do that.